File: 250507/숫자 2배 후 하나 제거하기/multiply-two-and-remove-one-number.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);
const arr = input[1].trim().split(' ').map(Number);
let min = Number.MAX_SAFE_INTEGER
for(let i = 0; i < n; i++){
    const temp = [...arr];
    let sum = 0
    temp[i] = temp[i] * 2;
    for(let j = 0; j < n; j++){
        const temp2 = [...temp]
        temp2.splice(j, 1)
        for(let j = 0; j < temp2.length - 1; j++){
            sum += Math.abs(temp2[j + 1] - temp2[j])
        }
        min = Math.min(sum, min)
    }

}
console.log(min)

File: 250812/숫자 2배 후 하나 제거하기/multiply-two-and-remove-one-number.py
import sys
n = int(input())
arr = list(map(int, input().split()))

# Please write your code here.
min_cnt = sys.maxsize
for i in range(n):
    temp = arr[:]
    temp[i] *= 2
    for j in range(n): 
        temp2 = [val for idx, val in enumerate(temp) if j != idx]
        cnt = 0
        for k in range(1, n-1):
            cnt += abs(temp2[k] - temp2[k-1])
        
        min_cnt = min(min_cnt, cnt)
        
print(min_cnt)